splunk-otel-android/src/main/java/com/splunk/rum/BandwidthTracker.java

```diff
@@ -0,0 +1,62 @@
+package com.splunk.rum;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class to track how much bandwidth is being used by span data.
+ * It tracks raw, uncompressed spans bytes being input to a sender and is not
+ * intended to represent actual gzipped/compressed/tls bytes on the network.
+ */
+class BandwidthTracker {
+    private final static int DATAPOINTS_TO_TRACK = 6;
+    private final Clock clock;
+    private final List<Long> times = new ArrayList<>();
+    private final List<Long> sizes = new ArrayList<>();
```

Comment:
That's probably a bit of micro-optimization, but WDYT about using `ArrayDeque` instead? It doesn't recreate the underlying array after removing the first element, it just changes the head index.